### ISSUE TITLE
What are the LineMOD results that should be obtained without changing the source code?

### ISSUE THREAD

**Issue by jpizarrom** [mentian/object-posenet#4]
Hello,
Thanks for sharing.

I would like to know if the results I am getting are correct, without doing changes to the code.
```
Test time 01h 43m 31s Epoch 49 TEST FINISH Avg dis: 0.013013, Accuracy: 0.858118
```

**Comment by mentian** [comment]
Which script are your running?
This result is not correct. To evaluate on LineMOD, you need to run eval_linemod.py. The accuracy should be close to 0.928 with provided model.

**Comment by jpizarrom** [comment]
I trained a new LineMOD model from scratch, using the code with the current hyperparameters.

**Comment by mentian** [comment]
I have trained the model multiple times. It should not deviate too much from the results reported in our paper.
Could you please make sure you exactly followed the steps in README? Could you also provide the results tested with our pre-trained model?

**Comment by jpizarrom** [comment]
### pre-trained model
```
python eval_linemod.py --model trained_models/linemod/pose_model_40_0.009487.pth \
    --output_result_dir results/eval_linemod_pose_model_40_0.009487
...
Object 1 success rate: 0.8541468064823642
Object 2 success rate: 0.9485935984481086
Object 4 success rate: 0.9137254901960784
Object 5 success rate: 0.9537401574803149
Object 6 success rate: 0.93812375249501
Object 8 success rate: 0.8136769078295342
Object 9 success rate: 0.8694835680751174
Object 10 success rate: 1.0
Object 11 success rate: 0.9980694980694981
Object 12 success rate: 0.9238820171265462
Object 13 success rate: 0.965270684371808
Object 14 success rate: 0.9376199616122841
Object 15 success rate: 0.9414024975984631
ALL success rate: 0.927484333034915
Accuracy: 0.9275180722911637
3 corrupted data
```

### trained models with https://github.com/mentian/object-posenet/tree/175c97a35f2ebed5994e98026c8911f2f32e6133
```
python eval_linemod.py --model results/linemod.2020080518/pose_model_49_0.013013.pth \
    --output_result_dir results/eval_linemod.2020080518_pose_model_49_0.013013
...
Object 1 success rate: 0.6978074356530028
Object 2 success rate: 0.8661493695441319
Object 4 success rate: 0.7803921568627451
Object 5 success rate: 0.8198818897637795
Object 6 success rate: 0.8862275449101796
Object 8 success rate: 0.7086223984142715
Object 9 success rate: 0.6967136150234742
Object 10 success rate: 1.0
Object 11 success rate: 0.997104247104247
Object 12 success rate: 0.7992388201712655
Object 13 success rate: 0.8794688457609806
Object 14 success rate: 0.8349328214971209
Object 15 success rate: 0.8443804034582133
ALL success rate: 0.831468218442256
Accuracy: 0.8316091960125701

```

```
python eval_linemod.py --model results/linemod.2020080519/pose_model_49_0.012852.pth \
    --output_result_dir results/eval_linemod.2020080519_pose_model_49_0.012852
...
Object 1 success rate: 0.6949475691134414
Object 2 success rate: 0.903976721629486
Object 4 success rate: 0.6715686274509803
Object 5 success rate: 0.8651574803149606
Object 6 success rate: 0.8772455089820359
Object 8 success rate: 0.7195242814667988
Object 9 success rate: 0.7887323943661971
Object 10 success rate: 1.0
Object 11 success rate: 0.9980694980694981
Object 12 success rate: 0.8477640342530923
Object 13 success rate: 0.797752808988764
Object 14 success rate: 0.9049904030710173
Object 15 success rate: 0.8904899135446686
ALL success rate: 0.8437033721277231
Accuracy: 0.8430937877885339
```

```
python eval_linemod.py --model results/linemod.2020080520/pose_model_49_0.013552.pth \
    --output_result_dir results/eval_linemod.2020080520/pose_model_49_0.013552
...
Object 1 success rate: 0.6196377502383222
Object 2 success rate: 0.8680892337536372
Object 4 success rate: 0.45588235294117646
Object 5 success rate: 0.7490157480314961
Object 6 success rate: 0.7864271457085829
Object 8 success rate: 0.6491575817641229
Object 9 success rate: 0.4863849765258216
Object 10 success rate: 0.9990592662276576
Object 11 success rate: 0.9961389961389961
Object 12 success rate: 0.6993339676498572
Object 13 success rate: 0.8621041879468846
Object 14 success rate: 0.800383877159309
Object 15 success rate: 0.8203650336215178
ALL success rate: 0.7529841838257236
Accuracy: 0.7532292398236448
```

### hw
```
bus            PRIME X370-PRO
memory         16GiB DIMM DDR4
processor      AMD Ryzen 5 2600 Six-Core Processor
graphics card  nVidia TU106 [GeForce RTX 2070]
```

**Comment by mentian** [comment]
I found one mistake [here](https://github.com/mentian/object-posenet/blob/2a647b3b17f78757154957c86de34f6936fe3e4f/train.py#L48).
opt.repeat_epoch should set to 10.
I apologize for the inconvenience caused. Could you try to train one more time? Let me know if it still has problem.
The evaluation result with our released model is correct.

**Comment by jpizarrom** [comment]
Thank you very much, this are the results with opt.repeat_epoch set to 10.

```
python eval_linemod.py --model results/linemod.2020080922/pose_model_49_0.010389.pth \
    --output_result_dir results/eval_linemod.2020080922/pose_model_49_0.010389
...
Object 1 success rate: 0.8188751191611058
Object 2 success rate: 0.938894277400582
Object 4 success rate: 0.8676470588235294
Object 5 success rate: 0.9291338582677166
Object 6 success rate: 0.9271457085828343
Object 8 success rate: 0.8067393458870169
Object 9 success rate: 0.8582159624413146
Object 10 success rate: 1.0
Object 11 success rate: 0.997104247104247
Object 12 success rate: 0.917221693625119
Object 13 success rate: 0.9366700715015321
Object 14 success rate: 0.8953934740882917
Object 15 success rate: 0.9365994236311239
ALL success rate: 0.9100268576544315
Accuracy: 0.9099723261934165
```
